File: java/src/main/java/lfs/step0_repl.java
package lfs;

import java.util.Scanner;


public class Zero {
    public static String READ(String str){
        return str;
    }

    public static String EVAL(String ast,String env){
        return ast;
    }

    public static String PRINT(String exp){
        return exp;
    }

    public static String RE(String env, String str){
        return EVAL(READ(str),env);
    }
    public static void main(String[] args) {

        if (args.length > 0) {
            return;
        }
        Scanner input = new Scanner(System.in);
        while (true) {
            String prompt = "==> ";
            String line =   input.nextLine();
            if (line == null) {
                continue;
            }
            System.out.println(PRINT(RE(null,line)));
        }
    }
}

File: golang/src/step0_repl/step0_repl.go
package main

import(
	"fmt"
	//"strings"
)

func  READ() string{
	var input string
	fmt.Print("\n-> ")
	fmt.Scanf("%s",&input)
	//add to history
	
	return input
}

func EVAL(ast string ,env string) string{
	return ast
}

func PRINT(exp string) string{
	return exp
}

func rep() string{
	return PRINT(EVAL(READ(),""))
}

func main(){

	fmt.Println("\v\t\t\t\t\tR E P L")
	for i := 0 ; i < 10 ; i++ {
		fmt.Print(rep())
	}
}
